File: October-17-2025/validParenthesis.js
var p = '()'

const validParenthesis = () => { 
  
  parenth = {
    '[': ']', 
    '{': '}', 
    '(': ')'
  }

  let stack = []

  for(let char of p){
    if(char in parenth){
      stack.push(parenth[char])
    } else { 
      if(stack.length === 0 || stack.pop() !== char){ 
        return false
      }
    }
  }
  return stack.length === 0
}

console.log(validParenthesis(p))

File: 2025/November-30-2025/validParenthesis.py




s = "()"

def validParenthesis(s: str) -> bool: 
  parenth = {
    '{':'}', 
    '(':')', 
    '[':']'
  }

  stack = []

  for ch in s: 
    if ch in parenth: 
      stack.append(parenth[ch])
    else: 
      if not stack: 
        return False
      last = stack.pop()
      if last != ch:
        return False
  return len(stack) == 0


print(validParenthesis(s))